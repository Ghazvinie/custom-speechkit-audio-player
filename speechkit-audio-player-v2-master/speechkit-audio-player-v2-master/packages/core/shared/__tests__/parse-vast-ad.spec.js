/* eslint-env jest */
import { xmlDoc } from './mocks/vast-xml-doc'
import processVastAd from '../ads/parse-vast-ad'

describe('parse-vast-ad', () => {
  let docs
  beforeEach(() => {
    docs = new DOMParser().parseFromString(xmlDoc, 'text/xml')
  })

  it('should parse xml document', () => {
    expect(processVastAd(docs)).toMatchSnapshot()
  })
})
