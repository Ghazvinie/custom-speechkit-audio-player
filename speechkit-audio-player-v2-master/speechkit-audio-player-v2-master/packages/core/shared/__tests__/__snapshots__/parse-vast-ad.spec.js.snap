// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse-vast-ad should parse xml document 1`] = `
Object {
  "data": Object {
    "campaign_id": "138312556267",
    "campaign_name": "Link",
    "id": "138312556267",
    "media": Array [
      Object {
        "altTitle": null,
        "bitrate": 0,
        "content_type": "application/x-mpegURL",
        "delivery": "streaming",
        "duration": 15000,
        "height": 0,
        "logo": "",
        "pricing": Object {
          "cost": 0,
          "currency": 0,
          "model": "CPM",
        },
        "promo_link": "https://pubads.g.doubleclick.net/pcs/click?xai=AKAOjsvEpomh_DmjBqV5roBomdosOcvK1U8Skoo9iT0yWZvQEtOhwdt6cece_wnAorWbHrIJQh2WjGgJGLrsVoVvqrGYXbKnRxthl6YTHFhtMHRG6hcLmsH3ZLyQ19WIyTjtrBYtEq9wRD8fx3gwRTczO9Cl28o55clIMRWZrkzG_hd3WJEZyfAoNBBm4byyikClb5wJu0J5Gt7fwoTyOO4VmwUWcW0saMNgUDeAt7qLnmPykirSgPPi6yJOPVyLRbfgCQcv3Xdr&sig=Cg0ArKJSzMxcASjjWjaQ&adurl=https://www.audible.co.uk/",
        "title": "Audible",
        "trackingUrls": Object {
          "complete": Array [
            "http://pubads.g.doubleclick.net/gampad/live/conversion/?ai=BDJ5z6z1-X_CXA8jJgQfSt4GoC7r0od5FAAAAEAEg9av9IDgAWOu9yKCDBGCEzfCFpB6yAQlsb2NhbGhvc3S6ARMzMzZ4MjY5LDQwMHgzMDBfeG1syAEF2gEtaHR0cDovL2xvY2FsaG9zdDo1MDAwL2Fkcy92YXN0LWdvb2dsZWFkcy5odG1sqQITt8M2DZm1PsACAuACAOoCFi8yMjAwMzkyMzQ2MS92YXN0YXVkaW_4AoTSHpADhAeYA4wGqAMB4AQB0gUGEL7Fj4QUkAYBoAYjqAfs1RuoB_PRG6gHltgbqAfC2hvYBwHgBx_SCAcIgGEQARgd2AgCgAoFmAsB&sigh=xDkDZqSMXkY&label=videoplaytime100&ad_mt=[AD_MT]",
          ],
          "firstQuartile": Array [
            "http://pubads.g.doubleclick.net/gampad/live/conversion/?ai=BDJ5z6z1-X_CXA8jJgQfSt4GoC7r0od5FAAAAEAEg9av9IDgAWOu9yKCDBGCEzfCFpB6yAQlsb2NhbGhvc3S6ARMzMzZ4MjY5LDQwMHgzMDBfeG1syAEF2gEtaHR0cDovL2xvY2FsaG9zdDo1MDAwL2Fkcy92YXN0LWdvb2dsZWFkcy5odG1sqQITt8M2DZm1PsACAuACAOoCFi8yMjAwMzkyMzQ2MS92YXN0YXVkaW_4AoTSHpADhAeYA4wGqAMB4AQB0gUGEL7Fj4QUkAYBoAYjqAfs1RuoB_PRG6gHltgbqAfC2hvYBwHgBx_SCAcIgGEQARgd2AgCgAoFmAsB&sigh=xDkDZqSMXkY&label=videoplaytime25&ad_mt=[AD_MT]",
          ],
          "impression": Array [
            "http://pubads.g.doubleclick.net/live/pcs/view?xai=AKAOjsubpqeJW3pK0vIBHDsm4KSLz_OTt29x--i3OkgE6l2NoQG1R_vwTHArGgD9JOner7nZfV_Sw_soV6r7DI9xuWt1lK1QCnTrWwrBMIf4cGH1Mg3Ozr97agwLP1g9606fyy8gKruRcgoXDEDAzoR8hDdF_O-9qAzhK7xCSn4vIQKQQhsHB8TSQzPp4qrlLr_eO1j8-QUtEWtqzp-hHDaPSDSwXEe_7ZWrcvpd58pHJfsOOHdIIK6_6GyUiY5ptrjiQ8KDmXwZKbxF&sig=Cg0ArKJSzDc2_ZjxbzZGEAE&adurl=",
          ],
          "midpoint": Array [
            "http://pubads.g.doubleclick.net/gampad/live/conversion/?ai=BDJ5z6z1-X_CXA8jJgQfSt4GoC7r0od5FAAAAEAEg9av9IDgAWOu9yKCDBGCEzfCFpB6yAQlsb2NhbGhvc3S6ARMzMzZ4MjY5LDQwMHgzMDBfeG1syAEF2gEtaHR0cDovL2xvY2FsaG9zdDo1MDAwL2Fkcy92YXN0LWdvb2dsZWFkcy5odG1sqQITt8M2DZm1PsACAuACAOoCFi8yMjAwMzkyMzQ2MS92YXN0YXVkaW_4AoTSHpADhAeYA4wGqAMB4AQB0gUGEL7Fj4QUkAYBoAYjqAfs1RuoB_PRG6gHltgbqAfC2hvYBwHgBx_SCAcIgGEQARgd2AgCgAoFmAsB&sigh=xDkDZqSMXkY&label=videoplaytime50&ad_mt=[AD_MT]",
          ],
          "start": Array [
            "http://pubads.g.doubleclick.net/gampad/live/conversion/?ai=BDJ5z6z1-X_CXA8jJgQfSt4GoC7r0od5FAAAAEAEg9av9IDgAWOu9yKCDBGCEzfCFpB6yAQlsb2NhbGhvc3S6ARMzMzZ4MjY5LDQwMHgzMDBfeG1syAEF2gEtaHR0cDovL2xvY2FsaG9zdDo1MDAwL2Fkcy92YXN0LWdvb2dsZWFkcy5odG1sqQITt8M2DZm1PsACAuACAOoCFi8yMjAwMzkyMzQ2MS92YXN0YXVkaW_4AoTSHpADhAeYA4wGqAMB4AQB0gUGEL7Fj4QUkAYBoAYjqAfs1RuoB_PRG6gHltgbqAfC2hvYBwHgBx_SCAcIgGEQARgd2AgCgAoFmAsB&sigh=xDkDZqSMXkY&label=part2viewed&ad_mt=[AD_MT]",
          ],
          "thirdQuartile": Array [
            "http://pubads.g.doubleclick.net/gampad/live/conversion/?ai=BDJ5z6z1-X_CXA8jJgQfSt4GoC7r0od5FAAAAEAEg9av9IDgAWOu9yKCDBGCEzfCFpB6yAQlsb2NhbGhvc3S6ARMzMzZ4MjY5LDQwMHgzMDBfeG1syAEF2gEtaHR0cDovL2xvY2FsaG9zdDo1MDAwL2Fkcy92YXN0LWdvb2dsZWFkcy5odG1sqQITt8M2DZm1PsACAuACAOoCFi8yMjAwMzkyMzQ2MS92YXN0YXVkaW_4AoTSHpADhAeYA4wGqAMB4AQB0gUGEL7Fj4QUkAYBoAYjqAfs1RuoB_PRG6gHltgbqAfC2hvYBwHgBx_SCAcIgGEQARgd2AgCgAoFmAsB&sigh=xDkDZqSMXkY&label=videoplaytime75&ad_mt=[AD_MT]",
          ],
        },
        "url": "http://redirector.gvt1.com/api/manifest/hls_variant/source/gfp_video_ads/id/429909428602d410/itag/0/playlist_type/LIVE/acao/yes/mime/application%2Fx-mpegURL/ctier/L/ip/0.0.0.0/ipbits/0/expire/1633644907/sparams/ip,ipbits,expire,source,id,itag,playlist_type,acao,mime,ctier/signature/7F4D819454B244D288FC3AE6238790E7B689D5BB.64C8A34D6D80570A26EF9C6C1A2B6D14935CCDD7/key/ck2/file/index.m3u8",
        "width": 0,
      },
    ],
  },
}
`;
